sdk/core/azure-core-management/src/main/java/com/azure/core/management/implementation/polling/SyncPollOperation.java

```diff
@@ -0,0 +1,170 @@
+            PollingState state = PollingState.from(serializerAdapter, pollingContext);
+            if (state.getOperationStatus().isComplete()) {
+                return pollResponseFromPollingState(serializerAdapter, pollResultType, state);
+            } else {
+                try (HttpResponse response
+                    = httpPipeline.sendSync(new HttpRequest(HttpMethod.GET, state.getPollUrl()), context)) {
+                    String body = response.getBodyAsBinaryData().toString();
+                    state.update(response.getStatusCode(), response.getHeaders(), body);
+                    state.store(pollingContext);
+                    return pollResponseFromPollingState(serializerAdapter, pollResultType, state);
+                }
+            }
+        };
+    }
+
+    /**
+     * Currently there is no option to cancel an ARM LRO in generic way, this is NOP.
+     *
+     * @param context the context
+     * @param <T> the type of poll result type
+     * @return cancel Function
+     */
+    public static <T> BiFunction<PollingContext<PollResult<T>>, PollResponse<PollResult<T>>, PollResult<T>>
+        cancelFunction(Context context) {
+        return (pollingContext, pollResultPollResponse) -> null;
+    }
+
+    /**
+     * Gets a Function that retrieves final result of a LRO.
+     *
+     * @param serializerAdapter the serializer for any encoding and decoding
+     * @param httpPipeline the HttpPipeline for fetching final result
+     * @param finalResultType the final result type
+     * @param context the context
+     * @param <T> the final result type
+     * @param <U> the poll result type
+     * @return retrieve final LRO result Function
+     */
+    public static <T, U> Function<PollingContext<T>, U> fetchResultFunction(SerializerAdapter serializerAdapter,
+        HttpPipeline httpPipeline, Type finalResultType, Context context) {
+        return pollingContext -> {
+            PollingState state = PollingState.from(serializerAdapter, pollingContext);
+            FinalResult finalResult = state.getFinalResult();
+            if (finalResult == null) {
+                return null;
+            } else {
+                String value = finalResult.getResult();
+                U result;
+                if (value != null) {
+                    result = PollOperation.deserialize(serializerAdapter, value, finalResultType);
+                } else {
+                    try (HttpResponse response
+                        = httpPipeline.sendSync(new HttpRequest(HttpMethod.GET, finalResult.getResultUri()), context)) {
+                        result = PollOperation.deserialize(serializerAdapter, response.getBodyAsBinaryData().toString(),
+                            finalResultType);
+                    }
+                }
+                return result;
+            }
+        };
+    }
+
+    private static <T> PollResponse<PollResult<T>> pollResponseFromPollingState(SerializerAdapter serializer,
```

Comment:
Are we able to extract this method to be shared?

We may actually does not need the "Mono" version of this function. I didn't see anything async in `PollOperation.pollResponseMonoFromPollingState`

Reply:
done